### PR TITLE
Add support for Full-text Search

### LINKED ISSUE #1374: Full-text Search
Add support for Full-text search as it is currently implemented in Core.

### PR DESCRIPTION
**Pull request by cmelchior**
Closes #1374 

This PR exposes the full-text search capabilities implemented in Core. Note the support is still very basic and only a simple token search is available.

**TODO**
- [x] Wait for https://github.com/realm/realm-core/pull/6517 to be merged.
- [x] Get a full description of supported functionality from Core
- [x] Extend tests to match what we write in the docs for this, so we can adjust the docs when/if the functionality changes.

### CODE REVIEW

File: packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/FullTextTests.kt [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                    result.exitCode,
+                    type.toString()
+                )
+                assertTrue(result.messages.contains(Regex("Full-text key .* is of type .* but must be of type")), result.messages)
+            }
+        }
+    }
+
+    @Test
+    fun fulltext_and_index_cannot_be_combined() {
```
Comment (clementetb) [inline packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/FullTextTests.kt:111]: FullText and primary key can be combined?

File: packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/annotations/Index.kt [hunk 2]
```diff
@@ -32,5 +32,7 @@ import org.mongodb.kbson.BsonObjectId
  * This annotation applies to the following primitive types: [String], [Boolean], [Byte], [Char],
  * [Short], [Int], [Long], [RealmInstant], [ObjectId], [BsonObjectId], [RealmUUID] as well as their
  * nullable variants.
+ *
+ * This annotation cannot be used at the same time as [FullText].
```
Comment (clementetb) [inline packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/annotations/Index.kt:36]:
```suggestion
 * This annotation cannot be combined with [FullText].
```

**Review by clementetb — APPROVED** [review]
Looks great, added some minor comments.

**Review by rorbech — APPROVED** [review]
Nice 🥳 - LGTM.

File: packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/FullTextTests.kt (hunk 1 above)
Comment (cmelchior) [reply packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/FullTextTests.kt:111]: No, that shouldn't be allowed. I have added a check preventing this now.

**Pull request merged**